File: app/src/main/java/nl/uva/datanose/scanner/DataNoseConnector.java
package nl.uva.datanose.scanner;


import android.util.Log;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;

import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

class DataNoseKeyResponse {
    String status;
    String message;

    public DataNoseKeyResponse(Response response) throws IOException, JSONException {

        JSONObject json = new JSONObject(response.body().string());

        status = json.getString("status");
        message = json.getString("message");
    }
}

class DataNoseCodeResponse {
    String status;
    String id;
    String student;
    String programme;
    String remarks;

    public DataNoseCodeResponse(Response response) throws IOException, JSONException {

        String body = response.body().string();
        Log.e("Got:", body);

        JSONObject json = new JSONObject(body);

        status = json.getString("status");
        id = json.getString("id");
        student = json.getString("student");
        programme = json.getString("programme");
        remarks = json.getString("remarks");
    }
}

public class DataNoseConnector {

    public static final String API_URL = "https://api-acc.datanose.nl/ScannerApp?key=";
    private OkHttpClient client;
    public String key;

    public DataNoseConnector(String key){
        client = new OkHttpClient();
        this.key = key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public DataNoseKeyResponse tryKey()
    {
        Request request = new Request.Builder().url(API_URL + key).build();

        try {
            return new DataNoseKeyResponse(client.newCall(request).execute());

        } catch (IOException e) {
            e.printStackTrace();

        } catch (JSONException e) {
            e.printStackTrace();
        }

        return null;
    }

    public DataNoseCodeResponse tryCode(String code) {
        Request request = new Request.Builder().url(API_URL + key + "&code=" + code).build();

        try {
            return new DataNoseCodeResponse(client.newCall(request).execute());

        } catch (IOException e) {
            e.printStackTrace();

        } catch (JSONException e) {
            e.printStackTrace();
        }

        return null;
    }

}

File: DataNoseScanner/DataNoseScanner/DataNose/DataNoseConnector.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Net.Http;
using System.Threading.Tasks;

namespace DataNoseScanner
{
    public class DataNoseConnector
    {
        //public const string API_URL = "https://api-acc.datanose.nl/ScannerApp?key="; // "https://api.datanose.nl/ScannerApp?key=";
        private string API_URL = "https://api-acc.datanose.nl/scannerapp";
        private HttpClient client;
        private ScannerAccount Account;

        public DataNoseConnector(string apiurl, ScannerAccount account)
        {
            api_url = apiurl;
            client = new HttpClient();
            client.Timeout = TimeSpan.FromSeconds(20);
            Account = account;
        }

        public DataNoseConnector(ScannerAccount account)
        {
            client = new HttpClient();
            client.Timeout = TimeSpan.FromSeconds(20);
            Account = account;
        }

        public string api_url
        {
            get { return API_URL; }
            set { API_URL = value; }
        }

        public ScannerAccount account
        {
            set { Account = value; }
        }

        public async Task<DataNoseKeyResponse> tryKey()
        {
            string request = API_URL + "/signup";
            string jsonString = Newtonsoft.Json.JsonConvert.SerializeObject(Account);
#if __ANDROID__
            Android.Util.Log.Info("webapi", request);
            Android.Util.Log.Info("webapi", jsonString);
#endif 
            try
            {
                HttpContent content = new StringContent(jsonString, Encoding.UTF8, "application/json");
                HttpResponseMessage response = await client.PostAsync(request, content);
                if (response.IsSuccessStatusCode)
                {
                    string body = await response.Content.ReadAsStringAsync();
#if __ANDROID__
                Android.Util.Log.Info("webapi", body);
#endif
                    return new DataNoseKeyResponse(body);
                }
                else
                {
#if __ANDROID__
                Android.Util.Log.Info("webapi", "post fail: " + response.StatusCode.ToString());
#endif
                }
            }
            catch
            {
                return null;
            }

            return null;
        }

        public async Task<DataNoseCodeResponse> tryCode(string code)
        {
            string request = API_URL + "/scan";
            ScannerScan scan = new ScannerScan() { Code = code, Account = Account };
            string jsonString = Newtonsoft.Json.JsonConvert.SerializeObject(scan);
#if __ANDROID__
            Android.Util.Log.Info("webapi", request);
            Android.Util.Log.Info("webapi", jsonString);
#endif 
            try
            {
                HttpContent content = new StringContent(jsonString, Encoding.UTF8, "application/json");
                HttpResponseMessage response = await client.PostAsync(request, content);
                if (response.IsSuccessStatusCode)
                {
                    string body = await response.Content.ReadAsStringAsync();
#if __ANDROID__
                Android.Util.Log.Info("webapi", body);
#endif
                    return new DataNoseCodeResponse(body);
                }
                else
                {
#if __ANDROID__
                Android.Util.Log.Info("webapi", "post fail: " + response.StatusCode.ToString());
#endif
                }
            }
            catch
            {
                return null;
            }

            return null;
        }

    }
}
